src/components/Faucet/Claim.tsx

```diff
@@ -1,34 +1,128 @@
-import { Button, Flex, Icon, useToast } from '@chakra-ui/react'
+import {
+  Button,
+  Flex,
+  Icon,
+  useToast,
+  Text,
+  Input,
+  FormControl,
+  FormHelperText,
+  FormErrorMessage,
+} from '@chakra-ui/react'
 import { ConnectButton } from '@rainbow-me/rainbowkit'
 import { errorToString, useClient } from '@vocdoni/react-providers'
 import { useEffect, useState } from 'react'
-import { useTranslation } from 'react-i18next'
-import { FaGithub } from 'react-icons/fa'
-import { useFaucet } from './use-faucet'
+import { Trans, useTranslation } from 'react-i18next'
+import { FaFacebook, FaGithub, FaGoogle } from 'react-icons/fa'
+import { signinUrlParams, useFaucet } from './use-faucet'
+import { ethers } from 'ethers'
+import { useForm } from 'react-hook-form'
+import { isValid } from 'date-fns'
+
+type ClaimProps = {
+  signinUrlParams?: signinUrlParams[]
+}
 
-export const Claim = () => {
+export const Claim = (props: ClaimProps) => {
   const { t } = useTranslation()
-  const { connected } = useClient()
+  const { account, connected, client } = useClient()
   const { loading, handleSignIn } = useClaim()
 
+  const [recipientAddress, setRecipientAddress] = useState<string>('')
+
+  const {
+    register,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      address: '',
+    },
+    mode: 'onChange',
+  })
+
+  useEffect(() => {
+    setRecipientAddress(account?.address as string)
+  }, [account])
+
+  const isValidAddress = async (address: string) => {
+    if (!ethers.utils.isAddress(address)) return false
+
+    let accountInfo
+    try {
+      accountInfo = await client.accountService.fetchAccountInfo(address)
+    } catch (error) {}
+
+    return !!accountInfo
+  }
+
+  const auth = (provider: string) => {
+    return handleSignIn(provider, recipientAddress, props.signinUrlParams || [])
+  }
```

Comment:
Considering `recipientAddress` is a state value, it woulud be safer to define it using `useCallback`:


```suggestion
  const auth = useCallback((provider: string) => 
    handleSignIn(provider, recipientAddress, props.signinUrlParams || [])
  , [recipientAddress])
```